src/ethereum/prague/exceptions.py

```diff
@@ -22,3 +22,63 @@ class TransactionTypeError(InvalidTransaction):
     def __init__(self, transaction_type: int):
         super().__init__(f"unknown transaction type `{transaction_type}`")
         self.transaction_type = transaction_type
+
+
+class TransactionTypeContractCreationError(InvalidTransaction):
+    """
+    Transaction type is not allowed for contract creation.
+    """
+
+    transaction_type: Final[int]
+    """
+    The type byte of the transaction that caused the error.
+    """
+
+    def __init__(self, transaction_type: int):
+        super().__init__(
+            f"transaction type `{transaction_type}` not allowed for "
+            "contract creation"
+        )
+        self.transaction_type = transaction_type
+
+
+class BlobGasLimitExceededError(InvalidTransaction):
+    """
+    The blob gas limit for the transaction exceeds the maximum allowed.
+    """
+
+
+class InsufficientMaxFeePerBlobGasError(InvalidTransaction):
+    """
+    The maximum fee per blob gas is insufficient for the transaction.
+    """
+
+
+class InsufficientMaxFeePerGasError(InvalidTransaction):
+    """
+    The maximum fee per gas is insufficient for the transaction.
+    """
+
```

Comment:
```suggestion
    def __init__(self, tx_max_fee_per_gas, block_base_fee_per_gas):
        super().__init__(f"insufficient max fee per gas ({tx_max_fee_per_gas} < {block_base_fee_per_gas})")
```

Could do something like this instead of passing in the message when constructing the instance?